### PR TITLE
FramePortal

### PR DESCRIPTION
**Pull request by tataihono**
- add FramePortal component
- add types for journeys-admin

### CODE REVIEW

File: apps/journeys-admin/src/components/FramePortal/FramePortal.tsx [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+  )
+
+  return (
+    <StylesProvider>
+      <CacheProvider value={cache}>{children}</CacheProvider>
+    </StylesProvider>
+  )
+}
+
+const StyledFrame = styled('iframe')(() => ({
```
Comment (storyworks) [inline apps/journeys-admin/src/components/FramePortal/FramePortal.tsx:41]: I'm going to add a todo for this for cooldown https://mui.com/system/styled/#how-to-use-components-selector-api

File: apps/journeys-admin/src/components/FramePortal/FramePortal.tsx [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+      createCache({
+        key: 'iframe',
+        container: document.head,
+        prepend: true
+      }),
+    [document]
+  )
+
+  return (
+    <StylesProvider>
```
Comment (storyworks) [inline apps/journeys-admin/src/components/FramePortal/FramePortal.tsx:35]: What's the use of `StylesProvider` if we're not passing any props to it atm? 
It's part of mui legacy styling so unless we really need it should we omit it? Removing it, the story still had correct styling..

File: apps/journeys-admin/src/components/FramePortal/FramePortal.tsx [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+      HTMLIFrameElement
+    >,
+    'css'
+  > {
+  children: ReactNode
+}
+
+export const FramePortal = memo(function FramePortal({
+  children,
+  ...other
```
Comment (storyworks) [inline apps/journeys-admin/src/components/FramePortal/FramePortal.tsx:58]:
```suggestion
  ...props
```
Shall we stick to React terminology?

File: apps/journeys-admin/src/components/FramePortal/FramePortal.tsx [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+}: FrameProps): ReactElement {
+  const frameRef = useRef<HTMLIFrameElement>(null)
+  // If we portal content into the iframe before the load event then that
+  // content is dropped in firefox.
+  const [iframeLoaded, onLoad] = useReducer(() => true, false)
```
Comment (storyworks) [inline apps/journeys-admin/src/components/FramePortal/FramePortal.tsx:63]:
```suggestion
  const [iframeLoaded, dispatch] = useReducer(() => true, false)
```
or use `dispatchLoad`? Keeping to React terminology makes it a bit more familiar to read + on___ comes with all sorts of other connotations.

File: apps/journeys-admin/apollo.config.js [hunk 5]
```diff
@@ -1,6 +1,6 @@
 module.exports = {
   client: {
-    includes: ['./pages/**', './src/**'],
+    includes: ['./pages/**', './src/**', '../../libs/journeys/ui/**'],
```
Comment (storyworks) [inline apps/journeys-admin/apollo.config.js:3]: What's this for?

**Review by storyworks — CHANGES_REQUESTED** [review]

File: apps/journeys-admin/src/components/FramePortal/FramePortal.tsx (hunk 3 above)
Comment (tataihono) [reply apps/journeys-admin/src/components/FramePortal/FramePortal.tsx:58]: is this the norm in react? I thought props was only used when it contained all the props. In react docs i've seen other and rest as the name of the remaining props.

File: apps/journeys-admin/apollo.config.js (hunk 5 above)
Comment (tataihono) [reply apps/journeys-admin/apollo.config.js:3]: needed to generate types with fragments in a different location.

File: apps/journeys-admin/src/components/FramePortal/FramePortal.tsx (hunk 2 above)
Comment (tataihono) [reply apps/journeys-admin/src/components/FramePortal/FramePortal.tsx]: removed

File: apps/journeys-admin/src/components/FramePortal/FramePortal.tsx (hunk 3 above)
Comment (tataihono) [reply apps/journeys-admin/src/components/FramePortal/FramePortal.tsx]: going with rest as I have seen that elsewhere.

File: apps/journeys-admin/src/components/FramePortal/FramePortal.tsx [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+      frameRef.current.contentDocument.readyState === 'complete' &&
+      !iframeLoaded
+    ) {
+      onLoad()
+    }
+  }, [iframeLoaded])
+
+  return (
+    <>
+      <StyledFrame onLoad={onLoad} ref={frameRef} {...other} />
```
[the remark this answers is not in the document]
Comment (tataihono) [reply apps/journeys-admin/src/components/FramePortal/FramePortal.tsx]: ok

File: apps/journeys-admin/src/components/FramePortal/FramePortal.tsx (hunk 4 above)
Comment (tataihono) [reply apps/journeys-admin/src/components/FramePortal/FramePortal.tsx]: done

File: apps/journeys-admin/src/components/FramePortal/FramePortal.spec.tsx [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+    )
+    const iframe = baseElement.getElementsByTagName('iframe')[0]
+    await waitFor(() =>
+      expect(
+        iframe.contentDocument?.body.getElementsByTagName('p')[0].innerHTML
+      ).toEqual('hello world')
+    )
+    expect(
+      iframe.contentDocument?.body.getElementsByTagName('p')[0]
+    ).toHaveStyle('font-family: "Roboto","Helvetica","Arial",sans-serif')
```
[the remark this answers is not in the document]
Comment (tataihono) [reply apps/journeys-admin/src/components/FramePortal/FramePortal.spec.tsx]: done

**Review by storyworks — CHANGES_REQUESTED** [review]
Merge in latest master? (It updates the config so we only use light theme in storybook - will change VR) Otherwise looks good :)

File: apps/journeys-admin/src/components/FramePortal/FramePortal.tsx (hunk 3 above)
Comment (storyworks) [reply apps/journeys-admin/src/components/FramePortal/FramePortal.tsx]: I've mainly just seen props

**Review by storyworks — APPROVED** [review]

**Pull request merged**